### ISSUE TITLE
Regarding hardware upgrades, such as type-c interface

### ISSUE THREAD

**Issue by GreenShadeZhang** [Open-Smartwatch/open-smartwatch-gps#12]
Is this project still being updated? I see that the hardware version is a long time ago. I want to reproduce a finished product, but the hardware feels a bit old, so I would like to ask if there is a timeline for this hardware update.

**Comment by a-allard** [comment]
@GreenShadeZhang 

This project has been quiet for quite some time.  As mentioned in #10 I forked this and started a new [revision of the hardware](https://github.com/a-allard/open-smartwatch-gps).  I have not yet tested it as the boards are finally in production.  Once I have them tested and get a fork of the OS running I'll update the fork of this repo with info on building.

**Comment by GreenShadeZhang** [comment]
> @GreenShadeZhang
> 
> This project has been quiet for quite some time. As mentioned in #10 I forked this and started a new [revision of the hardware](https://github.com/a-allard/open-smartwatch-gps). I have not yet tested it as the boards are finally in production. Once I have them tested and get a fork of the OS running I'll update the fork of this repo with info on building.

This sounds great, I think it would be a great toy if the GPS and circuit board worked properly, looking forward to your results.

**Comment by samuk** [comment]
How feasible would it be to do a variant based on https://store.rakwireless.com/products/wisduo-lpwan-module-rak11300?m=3&h=wisduo-modules-for-lorawan you'd lose wifi, but gain LoRa

**Comment by a-allard** [comment]
@samuk 

Such a variant looks to be possible.  But the FW would require a complete rewrite from the ground up.  The FW rewrite is not something I am up for.
As well as you'd be relying on always having a LoRa network to connect to.  As things are now most everyone has a WiFi network to connect to but I doubt the majority of people have access to LoRaWAN.  That said, battery life might, I stress might as RAK does not say anything about battery life at all, improve as LoRa tends to be lower power than WiFi.


So, I am not opposed to doing the HW design.  But someone else will have to write the FW for it.

**Comment by samuk** [comment]
That would be very interesting to me.

My use case would be using the watch as a basic watch & LoRa tracker. The firmware would essentially be https://meshtastic.org/docs/hardware/devices/rak/core-module?rakcore=RAK11310#rak11310---rp2040

The Twatch is already supported, but closed hardware and the ESP32 makes it power-hungry https://meshtastic.org/docs/hardware/devices/twatch/

With some additional UI stuff to tell the time. Meshtastic already has various UI's, including a clock, so I think it's fairly feasible.

**Comment by samuk** [comment]
![stm-rpi](https://github.com/Open-Smartwatch/open-smartwatch-gps/assets/400875/9a240274-c8b5-4da5-8c1b-4733f9a3702b)

What might actually make sense is designing a watch for the generic [Wisblock connector](https://store.rakwireless.com/products/wisblock-core-modules?variant=42440630304966)

It would then be up to users which module they used (STM, NRF52, RP2040, ESP32) if users wanted to use a variant of the existing firmware they could get the ESP32 module, or for Meshtastic based firmware one of the LoRa modules.

The Wisblock boards are a little larger, but might actually leave more space for routing on the watch PCB as they are separate from it?

**Comment by a-allard** [comment]
2 issues with using the generic Wisblock FP.

The display connector is 2.1mm high.  The Wisblock connector is only 0.8mm high.  So the Wisblock would collide with the display connector before it was correctly seated.  Even if you rotated the Wisblock 90 degrees so the connector was directly above the display connector the other end would stick beyond the edge of the watchy current size.

The other issue is when trying to add a module in the battery gets displaced this means the case gets significantly thicker.  Not the end of the world but it won't fit the watchy current case design.

Also the Wisblock has mounting screws to hold it in place.  Those are going to be an issue getting them in an area that does not harm the display.  Or leave them out and make the case hold the Wisblock with a tension fit which gets tricky due to mechanical tolerance stackup.


Watchy is nice and small because there are no modules on it and it does not support GPS.  To pull in a relatively large module (Wisblock) is going to swell the board beyond the current watchy outline.  The way to get a watchy sized device that supports LoRa would be to design a new version of Watchy with the guts of the RAK11310 directly laidout on the board and drop the GPS module and SD card from this design.  That would make doing something like a watchy sized device probably doable.


So, if you want a Wisblock module based watch and are willing to dump the GPS and SD card support it will still be larger than watchy but I can see what it looks like.  If you want the GPS and or SD card it is definitely going to be larger than watchy.

**Comment by samuk** [comment]
" Even if you rotated the Wisblock 90 degrees so the connector was directly above the display connector the other end would stick beyond the edge of the watchy current size." 

By my measure you _just_ get the required 20mm? It would be tight, but possible? The e-ink could even be brought down ~1mm lower maybe?
![20mm](https://github.com/Open-Smartwatch/open-smartwatch-gps/assets/400875/aab3e473-f3c9-4e6d-bfc0-19d6c12cf680)

I'd be happy to dump the SD personally, but the GPS is what would turn it into a useful tracker for my particular use case.

How large are the smallest (useful) GPS devices? Shall I do a bit of research?

**Comment by a-allard** [comment]
That might fit with a bit of shoehorning. The GPS will be crazy tight.

So all passives will be 0201s for space reasons. So hand assembly will be very difficult if possible. Not sure if you're planning to order fully assembled from pcbway or doing any hand assembly.

I won't have time to really look at this for a few weeks. But I might be able to do a course feasibility test.

**Comment by samuk** [comment]
That would be great. I'd use JLPCB assembly, so basic parts from LCSC would be a bonus.

If it did end up too fat for the cases we might get away with a spacer and longer screws?

**Comment by a-allard** [comment]
Alright. I'll have to remember what JLC limitations are.


That might be feasible. I don't know enough about how watchy mechanical assembly is designed to give a solid answer.

It'll be a bit getting this setup. When I get the new repo setup I'll ping you so you can follow as things progress provided the initial fit test looks good.

That said the WiFi version I started will take precedent when the boards come in.

**Comment by samuk** [comment]
Nice, thanks. I'll do a bit of research on GPS it's not an area I know much about. This[ MAX2769BETI/V+T ](https://www.lcsc.com/product-detail/Satellite-Positioning-Modules_Analog-Devices-Inc-Maxim-Integrated-MAX2769BETI-V-T_C118482.html) teensy for example, but not sure about codebase etc.

**Comment by Szybet** [comment]
Cool things, cool things
> You only get ~12cm below the e-ink connector to place a GPS module though. I don't know if that's adequate?

Below the e-ink connector goes the e-ink flex cable so...

well if you are looking for space on the PCB I would propose:
- Replace the battery charging circuit with something smaller
- Remove the accelerometer IC
- Remove the buzzer as already mentioned
- smaller battery connector?

**Comment by samuk** [comment]
The e-ink connector can pass over other components though?

One possibility is to abstract it away and provide a single RAK sensor connector. Users _might_ then choose to add the [10 x 23mm GPS module,](https://store.rakwireless.com/products/wisblock-gnss-location-module-rak12500?m=5&h=wisblock-sensor) or perhaps they'd prefer to clip in a [10x 10mm 9DOF IMU](https://store.rakwireless.com/products/9dof-motion-sensor-tdk-mpu9250-rak1905?m=5&h=wisblock-sensor)

**Comment by samuk** [comment]
Nice!

* Yes connecting directly would make sense I think
* I don't think 4layer is a particular concern? We're not necessarily making a product that has to be financially viable? 
* I don't have a part number for the Wisblock connector, I'll see if I can find one RAK do sell them though: https://www.shopiotstore.shop/product/rak-wireless-wisblock-wisconnector-8-pcs-rak-wireless/
* Using space under the RAK seems sensible
* Attached is a Kicad footprint for the smaller GPS 
[MAX2769BETI.zip](https://github.com/Open-Smartwatch/open-smartwatch-gps/files/14309227/MAX2769BETI.zip)
[MAX2769BETI/V+T]( https://www.lcsc.com/product-detail/Satellite-Positioning-Modules_Analog-Devices-Inc-Maxim-Integrated-MAX2769BETI-V-T_C118482.html) - not sure how the antennas work with them though.

My only concern with your mockup is USB/ Battery placement & height of the USB making Watchy cases impossible . I guess given the placement of the Wisblock module it's simply impossible to use the Wactchy MicroUSB placement.

Would small headers for the (irregular) programming and a [Wireless module](https://www.aliexpress.com/i/1005005617055228.html) be any thinner?

I can't open your project unfortunately, I'll try it again later

![errot](https://github.com/Open-Smartwatch/open-smartwatch-gps/assets/400875/62370f13-701e-4e76-a2b6-2141fa55702a)

**Comment by a-allard** [comment]
Sounds good.

I'll take a look at the other GPS module probably this weekend.

We can do a small header. Or because the wisblock is technically removable we could leave no programming connector on this board and one has to remove it to program.... less than ideal but it'd work.
Before we get there though let me think a bit. Might be able to do somthing like an edge connector.
We can do a wireless charger. It'll likely be either a separate stacking board and or an off the shelf module like the one you linked to.


Sorry the project won't open. I'm running kicad 8 (technically its a release candidate but the actual release is scheduled for Mar 2) its under the nightly builds for now. Kicad I was supposed to release end of January and it had a bunch of new features I really wanted to use. Sadly they changed the file format enough backwards comparability is completely broken. But old projects can be moves to kicad 8 just by opening them.

**Comment by samuk** [comment]
Good point about programming off-watch. I think that's probably the best (easiest!) option. If we're realistic this is probably a watch for a small subset of geeks anyway, so spending ~$9 on a [RAK board](https://store.rakwireless.com/products/wisblock-base-board-rak19003?m=5&h=wisblock-base) seems reasonable. 

So we're just left with the problem of getting power to the thing? 

Maybe we should target a particular watch case and design for that? The aluminum ones look nice, but I imagine they are an RF nightmare and not viable for any wireless charging or GPS/LoRa use?

I'll upgrade to Kicad8 later and have a proper look.

**Comment by a-allard** [comment]
Yeah. Charging is the last big hurdle before finalizing the HW and then layout.

I'm open to most any case. Aluminum is an RF nightmare. That said if front and back plates are plastic it not be an issue at all. But we won't know until after we build and test.....

**Comment by uvwxy** [comment]
Quick question, and sorry if I missed it in this thread:

Where will the GPS antenna go?

**Comment by a-allard** [comment]
Hi @uvwxy 

A bit of background, there are 2 different models being discussed here.  The first is the direct fork of your project [here](https://github.com/a-allard/open-smartwatch-gps).  The other is a complete redesign around the RAK modules.  The second is still early in HW dev but its [here](https://github.com/a-allard/OSW_rakWisblock)


For both designs the GPS has a built in antenna.  The direct fork I widened out the top quite a bit to better match what the manufacturer recommends for the GPS layout.  Not sure if it will work boards are still in China getting built right now.

For the RAK module I have not gotten into the datasheet enough to double check placement.  There is for sure a chance the GPS and LoRa will argue but the only way to find out is to build....
Thanks for the reminder though.

**Comment by samuk** [comment]
I'm assuming the [MAX2769BETI/V+T](https://www.lcsc.com/product-detail/Satellite-Positioning-Modules_Analog-Devices-Inc-Maxim-Integrated-MAX2769BETI-V-T_C118482.html) wouldn't have a built in antenna? So I guess it will be a judgement call as to whether the much smaller size makes it worth the bother. 

It might however fit _under_ the RAK module if the antenna is external

**Comment by a-allard** [comment]
Yeah the MAX chip likely doesn't have an antenna. The question at that point is do we go with an external antenna that plugs into the board or find a trace antenna that works.
The trouble with a trace antenna is they usually require a bit of tuning unless the design has already been tuned by someone else on a PCB similar enough.

Have to see.

As to fitting under... it would be a line to line fit and might hold the RAK module slightly (.2mm at most) above the standoffs. Where this isn't going to be a viable product that might be okay. I'll try to lay it out without doing that but if it comes to it that is an option.

**Comment by samuk** [comment]
I think my preferred option might be a [vertically mounted microusb](https://www.lcsc.com/product-detail/USB-Connectors_SHOU-HAN-MICRO-180-ZB_C668588.html)

I would mean we weren't 100% wactchy case compatible, but adapting existing 3d print cases would be fairly trivial, and it would mean that you could use a 3d print back piece and the [watchy aluminum frontpiece](https://shop.sqfmi.com/products/invader) assuming you don't care too much about RF.

**Comment by samuk** [comment]
In terms of the GPS antenna I wonder if we could use one from an [iphone](https://www.youtube.com/watch?v=iqbZRwuvUbs) an area of PCB under the RAK module could be cut away to correspond with the antenna placement?

For the LoRa the [RAK antenna](https://store.rakwireless.com/products/pcb-antenna-for-lora?variant=43045890326726) looks to be about 40mm long, the pigtail on it is far too long at 100mm, I'll see if I can find one with ~20mm

**Comment by samuk** [comment]
Nice! I still have concerns about the excess ribbon cable from the screen though?

Is a layout similar to the one I proposed not possible? The ribbon cable connector would probably need to be moved 'down' a mm or two to account for the components (GPS? Caps? Resistors?) that would go under the cable?

This [GPS1003 antenna](https://www.lcsc.com/product-detail/Antennas_Rainsun-microwave-Tech-GPS1003_C239243.html) might work for GPS?

**Comment by samuk** [comment]
![hacky](https://github.com/Open-Smartwatch/open-smartwatch-gps/assets/400875/08f1ee09-f82e-4319-99ad-2b586ba5f086)

If we're going to put the USB out the back, we do need to bear in mind the battery placement too, I think it has to go down 'below' the buttons, so that it doesn't foul the battery. T

**Comment by a-allard** [comment]
Looking at the MAX chip a bit more....  It will be very tight under the RAK.  The clearance might be non-existant.  RAK module sits ~0.72mm off the board (as near as I can tell from the CAD model for the mounting fixture they RAK gives  they aren't really clear about the height so that makes things difficult).  The MAX is 0.7-0.8mm tall.  So the RAK module might actually be sitting on top of the GPS chip.  Not ideal by any stretch.  But as long as the mounting screws are only snugged things should be okay....

In terms of an antenna the one you found has a massive recommended keep out area (49.5mm X 20mm).  I found a [molex](https://www.digikey.com/en/products/detail/molex/1462350001/6159253) antenna that is a bit taller, 4mm, but has a much smaller keep out.  I think it will work in the upper corner of the board with the MAX chip under the RAK module.

After rearranging things and adding a few more CAD models (RAK module and the display) here is what things are looking like:

![image](https://github.com/Open-Smartwatch/open-smartwatch-gps/assets/30708323/f068a46a-ed96-4dc5-9849-5dbfc081e325)

I tried to pull the LCD connector up some and pull the cable hole down some so the slack shouldn't be as big of an issue.  I did try to keep enough room between the connector and components in front of it so getting the cable into the connector won't be too miserable.
The battery connector is in a much better location for getting the battery to sit better.

Some height perspective.

| Side view | Angled view |
:---: | :---:
![image](https://github.com/Open-Smartwatch/open-smartwatch-gps/assets/30708323/381bc206-16fc-4cb8-ad65-bc645d3678f0) | ![image](https://github.com/Open-Smartwatch/open-smartwatch-gps/assets/30708323/30daceb3-65c1-408e-9936-1db40164bcef)

I will have to look at alternate USB options as the one currently in the model protrudes through the board enough I think it will mess with the display.  Other than that I need to finish wiring up the RAK module in the schematic and get to layout.

**Comment by samuk** [comment]
Nice, Looking good!

I had a look at my RAK device, it is hard to judge the 0.x mm. I _think_ it is 0.7mm though.
![hard](https://github.com/Open-Smartwatch/open-smartwatch-gps/assets/400875/70d3cb86-25c2-40a9-8589-352cb1ba1b05)

I think there are [narrower SMD vertical micro USB](https://www.mouser.co.uk/c/connectors/usb-connectors/?mounting%20angle=Vertical&product=Micro%20USB%20Type%20B%20Connectors&termination%20style=SMD%2FSMT) that might give us a bit more space to avoid fouling the watch strap in the bottom left corner.

Great progress though, I want one!

**Comment by a-allard** [comment]
Finally got some time to dig into the poorly documented GPS chip and get the schematic part pretty well taken care of.

![image](https://github.com/Open-Smartwatch/open-smartwatch-gps/assets/30708323/31787761-9ea0-4f48-acc4-ec85f59429f1)

Need to finish wiring the RAK module to the rest of the stuff then move on to layout.

Progress is slow but moving forward....

**Comment by samuk** [comment]
Nice! looking good

![invader-side](https://github.com/Open-Smartwatch/open-smartwatch-gps/assets/400875/335260e9-31bc-47fe-949a-8a21ea4f0255)
![invader](https://github.com/Open-Smartwatch/open-smartwatch-gps/assets/400875/15b7be42-7d98-43b6-82d2-5c2afe8eaf7a)

Looking at the case I'm still worried about how we make a strap attachment with the USB in that location. Is rotating the USB 90degrees a possibility? Or is it just too tight? Case file [here](https://github.com/sqfmi/watchy-cases/blob/main/invader/invader.step)

**Comment by a-allard** [comment]
Looking at the board in the case I don't think it is too bad.

![image](https://github.com/Open-Smartwatch/open-smartwatch-gps/assets/30708323/2c404752-7111-4182-9be7-3edd0cbb90f2)

It is tight for sure but it looks doable here.  In terms of moving to a horizontal connector, if can be done.  Something like [this](https://www.digikey.com/en/products/detail/amphenol-cs-fci/10118192-0002LF/6817756) would work.  The issue is the button would have to slide up ~3mm to make room for the USB connector as a horizontal one can't be placed on the curve like the vertical one can.

I'm okay either way.  Let me know which way you'd like to go.

Here is a step export of the layout as is in case you want to see how it drops into the case.
[oswRAKwisBlock3D.zip](https://github.com/Open-Smartwatch/open-smartwatch-gps/files/14658124/oswRAKwisBlock3D.zip)

Also, please give the schematic a look over.  I'm happy with it as is.  If you see any issue with the pinout of the RAK module or the GPS chip let me know so we don't have to do a respin for something foolish.  There is a PDF of the [schematic here](https://github.com/a-allard/OSW_rakWisblock/tree/master/Documention) in case kicad is not cooperating.

**Comment by samuk** [comment]
Nice progress, I note the buttons are out of alignment though.

![bottom-buttons](https://github.com/Open-Smartwatch/open-smartwatch-gps/assets/400875/e96f78c5-5f7c-4bac-a3c4-d99fb894d697)

Would moving the USB & Battery connector to the top edge be feasible? The buttons could be brought down into alignment then?

**Comment by a-allard** [comment]
A couple issues,

1. The oscillator is too tall to be put under the RAK module.
2. The GPS antenna will likely work well with the USB connector beside it.

The RAK module probably will not set correctly if the battery connector is there.  The battery connector will likely be under the module keeping it up too high.

If I pull the display connector down it might work

**Comment by samuk** [comment]
I wonder if there's an [alternate part](https://www.lcsc.com/products/Wire-To-Board-Wire-To-Wire-Connector_11068.html?keyword=jst%20ph) that might help the battery connector, perhaps a vertical might work to reduce the footprint?

I did have a good look at the schematic and didn't spot anything, but my skills in this area are fairly minimal

**Comment by a-allard** [comment]
A vertical battery connector will likely cause you problems with the case as the one you pointed out is 4.8 mm tall.  The front of the case to the back of the case is 4.3mm.  And that is before you plug in a battery which will make the connector larger.  So, it can be done but I doubt it will work out well.

Also, the oscillator Y1 cannot go under the RAK module.  The oscillator is 1mm tall.  Which will push the module up 0.3mm higher than it should be.  I think that will be enough to cause connection issues.


What do you think about this layout?

![image](https://github.com/Open-Smartwatch/open-smartwatch-gps/assets/30708323/94f93f2a-f5c4-4343-8ee6-cc7e4fa2d7d4)

If you want to check it out in KiCAD make sure you switch to the [altLayout branch](https://github.com/a-allard/OSW_rakWisblock/tree/altLayout)

I'm not a huge fan of it myself as the path to the antenna is rather long and might not be great.  But it puts the buttons in alignment with the ones on watchy.  It does move the display connector down slightly, and puts some of the LCD switcher supply components further away but it will probably be okay.

**Comment by samuk** [comment]
The main issue I see with your alt layout is the USB cable having to pass through the battery.

![lower-rak](https://github.com/Open-Smartwatch/open-smartwatch-gps/assets/400875/38d0fb24-7679-4a30-a00b-32f492901ae2)

How about something along these lines?

**Comment by a-allard** [comment]
All the regulators are too tall to be stuffed under the RAK.

If we swap to a horizontal USB connector do you think the alt layout will work?

**Comment by samuk** [comment]
It's only caps and resistors under the RAK? 

Or alternately your alt & horizontal USB might be the winner?

**Comment by a-allard** [comment]
Pretty much just caps and resistors.  The GPS chip is just small enough to fit under as well.

Alrighty.  I'll swap the USB connector out and see about getting it aligned with the Watchy one.  I might have to mirror the layout but that is not an issue.

Sounds like a plan!

**Comment by a-allard** [comment]
USB connector updated and moved to match watchy.  Everything else is rearranged so it looks decent.  I'll probably start routing tonight after one more check in the case to see everything lines up well.

**Comment by samuk** [comment]
One issue I just noticed is the way the cases attach assumes holes where we have deleted them

**Comment by a-allard** [comment]
That would make assembly a challenge. Lol.

I'll see what I can do. It looks like watchy uses 4. Putting 2 back should be easy. As to the other 2 I'll play with the layout later and let you know how it goes.

**Comment by Szybet** [comment]
Looks bulky, have you considered making a attachable extension on the side? I have done it with my battery
![image](https://github.com/Open-Smartwatch/open-smartwatch-gps/assets/53944559/73651251-7714-46c9-a817-ba4f0bf916c8)

**Comment by Szybet** [comment]
The watchy is too big, making it smaller in this iteration wouldn't hurt

> Is that a 3dprint battery case you have there?

Yes

**Comment by Szybet** [comment]
Also I would like to ask PLEASE add solder points for the esp32 direct serial connection, before the USB chip. The chip needs 5V to operate so in practice there is no debugging only on battery...

There is no need to keep problems for the past for the sake of compatibility, the same goes with the cases.

**Comment by samuk** [comment]
I think we're imagining that most programming/debugging will happen off-device. These [modules](https://store.rakwireless.com/collections/wisblock-core) (which may or may not be ESP32) can be dropped into one of the [existing carriers](https://store.rakwireless.com/collections/wisblock) for much software work. Some good test points are definitely desirable though, space permitting.

If we get to a working prototype based on Watchy dimensions I'd call that a win. My particular use case is mostly as an off-grid child tracker device. Last year I used a [T-echo](https://www.lilygo.cc/products/t-echo) so this 'watch' would already represent a significant miniaturization of that LoRa mesh functionality.

**Comment by a-allard** [comment]
@Szybet to get a smaller form factor for the watch would require a couple things, first not using the modules as direct PCB mounted HW usually is smaller than having modules.  The second thing would be a different display.  The size of watch is primarily dictated by the display.

The original goal of the RAK project is to be based on the module.  So going smaller is a hard no.

If you find a smaller display we can talk about another custom spin of watchy adding test points and shrinking the board (though I don't know how much we can actually shrink it without setting up a new project)

@samuk 
I'll get the holes put in tonight.  I wouldn't use the RAK module hole for mounting as it is pretty small but if it works and doesn't torque the module too much (I'd imagine you'll have to play with it to find out)
Do you have a particular hole size in mind?  Or just whatever watchy used?

**Comment by samuk** [comment]
It looks like the current 'slot' is 2mm, so that should work

![2mm](https://github.com/Open-Smartwatch/open-smartwatch-gps/assets/400875/4bdf184d-2aab-4be4-8654-3c6d637bbf07)

**Comment by a-allard** [comment]
@samuk 

Layout is done.  I might make a couple more tweaks in the next couple days.  Let me know if there is anything missing or wrong.  It is quite tight.  Once you've checked it and I've looked it over again I'll put in all the info needed to get it manufactured.

**Comment by samuk** [comment]
How exciting! Well done.

![push](https://github.com/Open-Smartwatch/open-smartwatch-gps/assets/400875/f3577877-167e-4f1f-94cb-8eaf4539c78f)

I think you need to do a push to github before I'll see it though.

**Comment by a-allard** [comment]
I forgot I was on the alt branch still.  Now updated on master you should be good.

**Comment by samuk** [comment]
Wow nice, It's looking great! I appreciate how much work that will have been..

![great](https://github.com/Open-Smartwatch/open-smartwatch-gps/assets/400875/9fa8491d-3598-4a67-8dca-e3845febb295)

I guess the next step is to make a couple and see if they actually work with the default Meshtastic firmware, I can then tweak the software from there for e-ink etc

**Comment by a-allard** [comment]
That's weird.  Fixed now though.  I updated the pdf schematic and exported an html BOM.  As well as put in the layout on the drawing layer the stackup and characteristics tables.  I have a copy in pcbway to see if they are okay with the design and if the BOM checks out.  it should but I wanted to double check.

I think you should be a go.  I'd advise going to pcbway as I have had better luck with them and they are not much more that jlcpcb last time I checked.  If you choose to use jlcpcb there are a few parts you'll have to figure out suitable substitutes for.  Once pcbway approves it I'll share the design so you can just order rather than having to manually enter all the parameters.

Otherwise let me know if anything else is missing.

**Comment by samuk** [comment]
Nice, waiting for PCBway review now, I ordered a case/strap/battery.

I documented the bit's I'm ordering here: https://openmeshwatch.github.io/ I sent you an invite to that organisation/repo if you want to make any changes to that page [here](https://github.com/openmeshwatch/openmeshwatch.github.io/blob/master/index.md)

**Comment by samuk** [comment]
![missingparts](https://github.com/Open-Smartwatch/open-smartwatch-gps/assets/400875/ae4075bd-45ae-4b99-bce5-8a115b53a807)

These are the parts that I'm struggling to find. In some cases I'm struggling to even find the original via google.

I'll keep trying with it, but if you're able to provide any of the missing Cxxx LCSC part numbers in the right column of this then that would be great https://github.com/openmeshwatch/OSW_rakWisblock/blob/master/production/bom.csv

**Comment by a-allard** [comment]
Sorry for the delay. Been a bit crazy with the solar eclipse.
I'll have a look later this week.

**Comment by samuk** [comment]
![unmatched](https://github.com/Open-Smartwatch/open-smartwatch-gps/assets/400875/963de6d2-e1cd-4cc6-8d7c-90276968ffad)

These are the remaining unmatched parts, I can try their global sourcing, but I expect that will be slow/expensive.

**Comment by a-allard** [comment]
* LCSC number for J1: C691131
* LCSC number for C31,C32: C5646770
* LCSC number for J5: 7407277

L2 I'll have to see if LCSC makes something similar.  This looks like it might work: https://www.lcsc.com/product-detail/Power-Inductors_SXN-Shun-Xiang-Nuo-Elec-SMNR4020-10UH_C135263.html  But it is a different size so they might not like the footprint.

**Comment by samuk** [comment]
Nice, thanks.

**Comment by a-allard** [comment]
R7: [C1512948](https://www.lcsc.com/product-detail/Chip-Resistor-Surface-Mount_PANASONIC-ERJ-XGNF1002Y_C1512948.html)
